### PR TITLE
Allow running lighthouse on other pages available in publish folder

### PR DESCRIPTION
**Pull request by aitchiss**
Allows users to target pages in the given directory other than `index.html`. Also removes the need to pass an empty string for `path` if targeting the top-level serve directory.

Example summary output

Closes https://github.com/netlify/pillar-workflow/issues/860

### CODE REVIEW

**Review by kevinsamoei — DISMISSED** [review]
LGTM!

**Review by jackbrewer — CHANGES_REQUESTED** [review]
Code looks great ☺️ 

Regarding the README, I feel like the reasoning for having this new option isn't explained as much as it could be, and the option name could maybe be tweaked (disclaimer, it's still confusing to me!)

I thought this work was achieving something different than being able to directly address (html) files, although this work does also fix the problem I though we were aiming for so it's all looking good!

**Example**

Before: [This 11ty site deploy](https://app.netlify.com/sites/lighthouse-pwa-icon-test/deploys/633fef15e77faf00081ce1e8) – if you go to the _second_ lighthouse report and look at the report, it mentions missing styles/scripts etc. Also note the scores aren't great. This was run with `path = posts/`. As express is serving the `/posts` page directly _as the site root_ (this is what the path currently means), links to assets outside the posts directory break.

After: [This follow-up deploy](https://app.netlify.com/sites/lighthouse-pwa-icon-test/deploys/63400828d73ce80008511902) was run with `fileName = posts/` (no path set), and the second report in that deploy now has all assets available, the screenshots look fixed, and the scores are back up where they should be. Express is running the site from the root directory, then we're looking into the sites `posts` directory using fileName. All the asset paths are correct and reachable as the site is running from the root directory. A valid `fileName` value could be `posts/category/date/title`, which doesn't fit too well with being a "file name"

I _think_:
- path = where should I start up the server
- fileName = from path, where shall I look for the actual file to test

**Comment by jackbrewer** [comment]
Urgh, I meant to comment, not reject. Happy to resubmit as approved if you'd rather update docs etc as a follow-on

**Comment by jackbrewer** [comment]
Now I think about it more, should we be de-emphasising the use of `path`, and making `filePath` the primary option mentioned in README/docs. I imagine it's a lot more useful in most cases, and actually does what most people assume `path` does.

**Comment by aitchiss** [comment]
This is all great context thank you @jackbrewer!

> I think:
>
> path = where should I start up the server
> fileName = from path, where shall I look for the actual file to test – does "fileName" still work here?

Yes I think this is the right distinction, and no I don't think the current names reflect it well 😂

What I would _like_ to change these to is:

```
serveDir = what to serve
path = where to look inside to test - could be a dir, could be a specific html file
```

What do you think? It feels a bit wrong that `path` is being shifted to a new meaning, but I don't think there's any negative consequence of shifting doing that (and updating the documentation to match).

**Comment by nasivuela** [comment]
> What I would _like_ to change these to is:
> 
> ```
> serveDir = what to serve
> path = where to look inside to test - could be a dir, could be a specific html file
> ```
> 
> What do you think? It feels a bit wrong that `path` is being shifted to a new meaning, but I don't think there's any negative consequence of doing that (and updating the documentation to match). If anything, it's bringing things more inline with what you'd expect (like the 11ty example you linked above).
> 
> `serveDir` would probably become a less-used option.

I also think this approach is clearer!
Changing the meaning of `path` will be an improvement. As long as we document it and bump the mayor version to reflect the breaking change I do not see any negative consequence.

**Review by jackbrewer — APPROVED** [review]
Looking good 😄

File: src/index.js [hunk 1]
```diff
@@ -273,13 +275,16 @@ module.exports = {
         inputs,
       });
 
+      console.log('------', audits);
```
Comment (jackbrewer) [inline src/index.js:278]: @aitchiss I noticed this appearing in the deploy logs, not sure if it was intentional?! 🙃

Comment (aitchiss) [reply src/index.js:278]: 🙈

**Pull request merged**